nvidia-texture-tools fails to build on aarch64

Build fails with following:

```
-- Build files have been written to: /build/source/build
building
build flags: -j64 -l64 SHELL=/nix/store/nvilcbwjh6dq1ajm2r6ndk103c4zwryy-bash-4.4-p23/bin/bash
Scanning dependencies of target posh
Scanning dependencies of target rg_etc1
Scanning dependencies of target butteraugli
Scanning dependencies of target squish
Scanning dependencies of target nvcore
Scanning dependencies of target EtcLib
[  1%] Building C object extern/poshlib/CMakeFiles/posh.dir/posh.c.o
[  3%] Building CXX object extern/butteraugli/CMakeFiles/butteraugli.dir/butteraugli.cc.o
[  3%] Building CXX object extern/rg_etc1_v104/CMakeFiles/rg_etc1.dir/rg_etc1.cpp.o
[  4%] Building CXX object src/nvtt/squish/CMakeFiles/squish.dir/fastclusterfit.cpp.o
[  4%] Building CXX object src/nvtt/squish/CMakeFiles/squish.dir/colourblock.cpp.o
[  5%] Building CXX object src/nvtt/squish/CMakeFiles/squish.dir/weightedclusterfit.cpp.o
[  6%] Building CXX object src/nvtt/squish/CMakeFiles/squish.dir/colourfit.cpp.o
[  8%] Building CXX object src/nvtt/squish/CMakeFiles/squish.dir/maths.cpp.o
[  8%] Building CXX object src/nvtt/squish/CMakeFiles/squish.dir/colourset.cpp.o
[  8%] Building CXX object src/nvcore/CMakeFiles/nvcore.dir/Debug.cpp.o
[  9%] Building CXX object src/nvcore/CMakeFiles/nvcore.dir/FileSystem.cpp.o
[ 10%] Building CXX object src/nvcore/CMakeFiles/nvcore.dir/Memory.cpp.o
[ 11%] Building CXX object src/nvcore/CMakeFiles/nvcore.dir/StrLib.cpp.o
[ 12%] Building CXX object src/nvcore/CMakeFiles/nvcore.dir/Timer.cpp.o
[ 12%] Building CXX object src/nvcore/CMakeFiles/nvcore.dir/TextWriter.cpp.o
[ 12%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/Etc/Etc.cpp.o
[ 13%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/Etc/EtcImage.cpp.o
[ 14%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcBlock4x4Encoding_ETC1.cpp.o
[ 16%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcBlock4x4Encoding_R11.cpp.o
[ 16%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/Etc/EtcMath.cpp.o
[ 16%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcBlock4x4.cpp.o
[ 17%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcBlock4x4Encoding.cpp.o
[ 19%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcBlock4x4Encoding_RGB8.cpp.o
[ 19%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcBlock4x4Encoding_RG11.cpp.o
[ 20%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcBlock4x4Encoding_RGB8A1.cpp.o
[ 23%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcBlock4x4Encoding_RGBA8.cpp.o
[ 23%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcDifferentialTrys.cpp.o
[ 23%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcIndividualTrys.cpp.o
[ 23%] Building CXX object extern/EtcLib/CMakeFiles/EtcLib.dir/EtcCodec/EtcSortedBlockList.cpp.o
In file included from /build/source/src/nvcore/Stream.h:8:0,
                 from /build/source/src/nvcore/TextWriter.h:8,
                 from /build/source/src/nvcore/TextWriter.cpp:3:
/build/source/src/nvcore/Debug.h: In function 'bool nv::isValidPtr(const void*)':
/build/source/src/nvcore/Debug.h:174:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcccccccc) return false;
                                      ^
/build/source/src/nvcore/Debug.h:175:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcdcdcdcd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:176:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xdddddddd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:177:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xffffffff) return false;
                                      ^
make[2]: *** [src/nvcore/CMakeFiles/nvcore.dir/build.make:115: src/nvcore/CMakeFiles/nvcore.dir/TextWriter.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 24%] Linking C static library libposh.a
[ 24%] Built target posh
/build/source/src/nvcore/Debug.cpp:672:8: error: #error "Unknown CPU"
 #      error "Unknown CPU"
        ^~~~~
In file included from /build/source/src/nvcore/Debug.cpp:3:0:
/build/source/src/nvcore/Debug.h: In function 'bool nv::isValidPtr(const void*)':
/build/source/src/nvcore/Debug.h:174:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcccccccc) return false;
                                      ^
/build/source/src/nvcore/Debug.h:175:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcdcdcdcd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:176:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xdddddddd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:177:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xffffffff) return false;
                                      ^
In file included from /build/source/src/nvcore/Memory.cpp:4:0:
/build/source/src/nvcore/Debug.h: In function 'bool nv::isValidPtr(const void*)':
/build/source/src/nvcore/Debug.h:174:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcccccccc) return false;
                                      ^
/build/source/src/nvcore/Debug.h:175:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcdcdcdcd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:176:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xdddddddd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:177:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xffffffff) return false;
                                      ^
In file included from /build/source/src/nvcore/Stream.h:8:0,
                 from /build/source/src/nvcore/StdStream.h:8,
                 from /build/source/src/nvcore/FileSystem.cpp:78:
/build/source/src/nvcore/Debug.h: In function 'bool nv::isValidPtr(const void*)':
/build/source/src/nvcore/Debug.h:174:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcccccccc) return false;
                                      ^
In file included from /build/source/src/nvcore/StrLib.h:7:0,
                 from /build/source/src/nvcore/StrLib.cpp:3:
/build/source/src/nvcore/Debug.h: In function 'bool nv::isValidPtr(const void*)':
/build/source/src/nvcore/Debug.h:174:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcccccccc) return false;
                                      ^
/build/source/src/nvcore/Debug.h:175:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcdcdcdcd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:176:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xdddddddd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:175:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xcdcdcdcd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:177:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xffffffff) return false;
                                      ^
/build/source/src/nvcore/Debug.h:176:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xdddddddd) return false;
                                      ^
/build/source/src/nvcore/Debug.h:177:38: error: cast from 'const void*' to 'uint32 {aka unsigned int}' loses precision [-fpermissive]
      if (reinterpret_cast<uint32>(ptr) == 0xffffffff) return false;
                                      ^
make[2]: *** [src/nvcore/CMakeFiles/nvcore.dir/build.make:89: src/nvcore/CMakeFiles/nvcore.dir/Memory.cpp.o] Error 1
make[2]: *** [src/nvcore/CMakeFiles/nvcore.dir/build.make:76: src/nvcore/CMakeFiles/nvcore.dir/FileSystem.cpp.o] Error 1
/build/source/src/nvcore/Debug.cpp: In function 'void nv::debug::dumpCallstack(nv::MessageHandler*, int)':
/build/source/src/nvcore/Debug.cpp:1024:47: error: no matching function for call to 'nv::MessageHandler::log(const char*&, NULL)'
             messageHandler->log(lines[i], NULL);
                                               ^
In file included from /build/source/src/nvcore/Debug.cpp:3:0:
/build/source/src/nvcore/Debug.h:184:22: note: candidate: virtual void nv::MessageHandler::log(const char*, va_list)
         virtual void log(const char * str, va_list arg) = 0;
                      ^~~
/build/source/src/nvcore/Debug.h:184:22: note:   no known conversion for argument 2 from 'long int' to 'va_list {aka __va_list}'
make[2]: *** [src/nvcore/CMakeFiles/nvcore.dir/build.make:63: src/nvcore/CMakeFiles/nvcore.dir/Debug.cpp.o] Error 1
/build/source/src/nvcore/StrLib.cpp: At global scope:
/build/source/src/nvcore/StrLib.cpp:603:6: warning: 'void nv::swap(nv::StringBuilder&, nv::StringBuilder&)' has not been declared within nv
 void nv::swap(StringBuilder & a, StringBuilder & b) {
      ^~
In file included from /build/source/src/nvcore/StrLib.cpp:3:0:
/build/source/src/nvcore/StrLib.h:172:21: note: only here as a friend
         friend void swap(StringBuilder & a, StringBuilder & b);
                     ^~~~
/build/source/src/nvcore/StrLib.cpp:807:6: warning: 'void nv::swap(nv::String&, nv::String&)' has not been declared within nv
 void nv::swap(String & a, String & b) {
      ^~
In file included from /build/source/src/nvcore/StrLib.cpp:3:0:
/build/source/src/nvcore/StrLib.h:361:21: note: only here as a friend
         friend void swap(String & a, String & b);
                     ^~~~
make[2]: *** [src/nvcore/CMakeFiles/nvcore.dir/build.make:102: src/nvcore/CMakeFiles/nvcore.dir/StrLib.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:374: src/nvcore/CMakeFiles/nvcore.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 25%] Linking CXX static library libsquish.a
[ 25%] Built target squish
[ 26%] Linking CXX static library librg_etc1.a
[ 26%] Built target rg_etc1
[ 27%] Linking CXX static library libEtcLib.a
[ 27%] Built target EtcLib
[ 28%] Linking CXX static library libbutteraugli.a
[ 28%] Built target butteraugli
make: *** [Makefile:163: all] Error 2```

--- a/src/nvcore/Debug.cpp
+++ b/src/nvcore/Debug.cpp
@@ -1015,6 +1015,12 @@ void debug::dumpInfo()
 #endif
 }
 
+static void getEmptyVAList(va_list& list, ...)
+{
+    va_start(list, list);
+    va_end(list);
+}
+
 /// Dump callstack using the specified handler.
 void debug::dumpCallstack(MessageHandler *messageHandler, int callstackLevelsToSkip /*= 0*/)
 {
@@ -1027,8 +1033,11 @@ void debug::dumpCallstack(MessageHandler *messageHandler, int callstackLevelsToS
         Array<const char *> lines;
         writeStackTrace(trace, size, callstackLevelsToSkip + 1, lines);     // + 1 to skip the call to dumpCallstack
 
+        va_list empty;
+        getEmptyVAList(empty);
+
         for (uint i = 0; i < lines.count(); i++) {
-            messageHandler->log(lines[i], NULL);
+            messageHandler->log(lines[i], empty);
             delete lines[i];
         }
     }

--- a/src/nvcore/Debug.h
+++ b/src/nvcore/Debug.h
@@ -166,7 +166,7 @@ NVCORE_API void NV_CDECL nvDebugPrint( const char *msg, ... ) __attribute__((for
 namespace nv
 {
     inline bool isValidPtr(const void * ptr) {
-    #if NV_CPU_X86_64 || POSH_CPU_PPC64
+    #if NV_CPU_X86_64 || POSH_CPU_PPC64 || NV_CPU_AARCH64
         if (ptr == NULL) return true;
         if (reinterpret_cast<uint64>(ptr) < 0x10000ULL) return false;
         if (reinterpret_cast<uint64>(ptr) >= 0x000007FFFFFEFFFFULL) return false;

